### PR TITLE
obfuscate: init at unstable

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

More apps

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112324` at 2f2d08f5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>obfuscate</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/build-support/rust/default.nix:209:3:
```
    |
209 |   buildPhase = with builtins; args.buildPhase or ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/build-support/rust/default.nix:259:3:
```
    |
259 |   installPhase = args.installPhase or ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/obfuscate/default.nix:63:5:
```
   |
63 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112324` at 2f2d08f5 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>obfuscate</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/build-support/rust/default.nix:209:3:
```
    |
209 |   buildPhase = with builtins; args.buildPhase or ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/build-support/rust/default.nix:259:3:
```
    |
259 |   installPhase = args.installPhase or ''
    |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/obfuscate/default.nix:63:5:
```
   |
63 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/obfuscate/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+, rustPlatform
+, desktop-file-utils
+, appstream-glib
+, python3
+, librsvg
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "obfuscate";
+  version = "unstable"; # latest tag v0.0.2 is broken
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/obfuscate/default.nix:20]:
```suggestion
  version = "unstable-XXXX-XX-XX"; # latest tag v0.0.2 is broken
```

File: pkgs/applications/misc/obfuscate/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    glib
+    librsvg
+    gdk-pixbuf
+  ];
+
+  buildPhase = ''
+    patchShebangs build-aux/meson_post_install.py
+    mesonConfigurePhase
+    ninjaBuildPhase
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/obfuscate/default.nix:54]:
```suggestion
  preBuild = ''
    patchShebangs build-aux/meson_post_install.py
  '';
```

File: pkgs/applications/misc/obfuscate/default.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  installPhase = ''
+    ninjaInstallPhase
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/obfuscate/default.nix:58]:
```suggestion
```

I am not sure what you are trying to achieve.

File: pkgs/applications/misc/obfuscate/default.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  meta = with lib; {
+    description = "Censor private information";
+    homepage = "https://gitlab.gnome.org/World/obfuscate";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/obfuscate/default.nix:63]: gpl3Only gpl3Plus?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/obfuscate/default.nix [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "obfuscate";
+  version = "unstable"; # latest tag v0.0.2 is broken
+
+  src = fetchgit {
+    url = "https://gitlab.gnome.org/World/obfuscate.git";
+    rev = "1ec7e2d2b1a78c15a98cdb76be6a1d285accfa54";
+    sha256 = "1v58ma9gb2nrin2d2q2ww2qdhyd51blldnycljpf0qf1j6zj83cp";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/obfuscate/default.nix:26]: Please use fetchFromGitLab.

File: pkgs/applications/misc/obfuscate/default.nix (hunk 2 above)
Comment (mkg20001) [reply pkgs/applications/misc/obfuscate/default.nix:58]: The project uses meson to build the rust code and so this hackery is necesarry to override the rust hooks

File: pkgs/applications/misc/obfuscate/default.nix (hunk 3 above)
Comment (mkg20001) [reply pkgs/applications/misc/obfuscate/default.nix:62]: Overriding the rust phase that would usually run here

**Pull request closed without merging**